### PR TITLE
New package: hatch-1.2.1

### LINKED ISSUE #37334: [Package request] python hatch
Python packages `hatch` and `hatchling` are increasingly needed to package modern python packages.

### PR DESCRIPTION
**Pull request by jcgruenhage**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### Bikeshedding opportunities
 - ~~As they aren't just libraries, but also CLI tools, do hatch/hatchling need the `python3-` prefix? I personally don't like it, and would argue in favor of removing it.~~ Considering no one commented on this bit, I've decided to just get rid of.
 - ~~There's nearly no testing happening here, but that's due to the deps either requiring git checkouts with submodules, or themselves to be installed for testing. Any opinions on this? I'd prefer to have some amount of automatic testing here and not just a "JC has run this on his laptop once", considering that these are likely to be fairly central Python packages~~ there are a few tests failing now, but they are being run and there's a hundred times more tests that do work as intended. The ones failing where it's expecting to be running /usr/bin/python, but is actually running /usr/bin/python3 or similar things, which is fine IMO.
 - ~~Is a major update of python3-click something that might need additional testing? 29 of our packages depend on it, but on the other hand, it's *probably* fine, right?~~ Has been updated outside of this PR.
 
Fixes #37334

### CODE REVIEW

File: srcpkgs/python3-hatch/template [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+makedepends=""
+depends="python3-atomicwrites python3-click python3-hatchling python3-httpx python3-keyring python3-pexpect python3-platformdirs python3-pyperclip python3-rich python3-tomli-w python3-tomlkit python3-userpath python3-virtualenv"
+short_desc="Modern, extensible Python project management "
+maintainer="Jan Christian Grünhage <jan.christian@gruenhage.xyz>"
+license="MIT"
+homepage="https://hatch.pypa.io/latest/"
+distfiles="${PYPI_SITE}/h/${pkgname/python3-/}/${pkgname/python3-/}-${version}.tar.gz"
+checksum=61761e1fe14474fb2be7e264555b99fc343b4e63c0ee32aa3497c651e4cdec03
+# Requires itself to be installed for testing
+make_check=no
```
Comment (Chocimier) [inline srcpkgs/python3-hatch/template:24]: It only requires being on import path: `make_check_pre="env PYTHONPATH=./src"`

Comment (jcgruenhage) [reply srcpkgs/python3-hatch/template]: Indeed! With this added (and git added to `checkdepends` it runs mot of the tests now :)

File: srcpkgs/python3-tomlkit/template [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+depends="python3"
+short_desc="Style-preserving TOML library for Python"
+maintainer="Jan Christian Grünhage <jan.christian@gruenhage.xyz>"
+license="MIT"
+homepage="https://github.com/sdispater/tomlkit"
+changelog="https://github.com/sdispater/tomlkit/raw/master/CHANGELOG.md"
+distfiles="${PYPI_SITE}/t/${pkgname/python3-/}/${pkgname/python3-/}-${version}.tar.gz"
+checksum=71ceb10c0eefd8b8f11fe34e8a51ad07812cb1dc3de23247425fbc9ddc47b9dd
+#Requires a git submodule for running tests
+make_check=no
```
Comment (paper42) [inline srcpkgs/python3-tomlkit/template:17]: Could you explain why you think this package requires a git submodule for running tests? I had the package ready for a pylint update that I didn't merge yet, so I pushed my template with tests working: 72d051fb7312c60bbb05fc96f9df4662d99e7c90.

Comment (jcgruenhage) [reply srcpkgs/python3-tomlkit/template:17]: It's what upstream claims, see https://github.com/sdispater/tomlkit/#running-tests

I've just seen your merged template in the notifications, and am rebasing. Happy  to hear that apparently we can run the tests there.

Comment (paper42) [reply srcpkgs/python3-tomlkit/template:17]: But the PyPI tarballs have that submodule bundled inside. It's always good to try to run tests before looking upstream.

**Comment by icp1994** [comment]
Hello, I didn't you were working on this already. I have been trying to package this for a few days and finally managed to get a working build. Since you have this PR already, please feel free to reuse any part of [my commit](https://github.com/icp1994/void-packages/commit/a75b698c88d153464d6cda9be1b19be7cb738107) if you wish.

File: srcpkgs/python3-click/template [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Template file for 'python3-click'
 pkgname=python3-click
-version=7.1.2
-revision=5
+version=8.1.3
+revision=1
 wrksrc="click-${version}"
 build_style="python3-module"
 hostmakedepends="python3-setuptools"
```
Comment (paper42) [inline srcpkgs/python3-click/template:7]: add python3-pytest to checkdepends to actually run tests

File: srcpkgs/python3-click/template [hunk 4]
```diff
@@ -11,7 +11,7 @@ maintainer="Oliver Kiddle <okiddle@yahoo.co.uk>"
 license="BSD-3-Clause"
 homepage="http://click.pocoo.org/"
```
Comment (paper42) [inline srcpkgs/python3-click/template:12]:
```suggestion
homepage="https://palletsprojects.com/p/click/"
changelog="https://raw.githubusercontent.com/pallets/click/main/CHANGES.rst"
```

**Comment by leahneukirchen** [comment]
Please rebase.

**Pull request merged**